specification/servicenetworking/resource-manager/Microsoft.ServiceNetworking/preview/2024-05-01-preview/TrafficController.json

```diff
@@ -0,0 +1,1895 @@
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "trafficControllerName",
+            "in": "path",
+            "description": "traffic controller name for path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[A-Za-z0-9]([A-Za-z0-9-_.]{0,62}[A-Za-z0-9])?$"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Azure operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/SecurityPolicyListResult"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Security Policies": {
+            "$ref": "./examples/SecurityPoliciesGet.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceNetworking/trafficControllers/{trafficControllerName}/securityPolicy/{securityPolicyName}": {
+      "get": {
+        "operationId": "SecurityPoliciesInterface_Get",
+        "tags": [
+          "SecurityPoliciesInterface"
+        ],
+        "description": "Get a SecurityPolicy",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "trafficControllerName",
+            "in": "path",
+            "description": "traffic controller name for path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[A-Za-z0-9]([A-Za-z0-9-_.]{0,62}[A-Za-z0-9])?$"
+          },
+          {
+            "name": "securityPolicyName",
+            "in": "path",
+            "description": "SecurityPolicy name for path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[A-Za-z0-9]([A-Za-z0-9-_.]{0,62}[A-Za-z0-9])?$"
+          }
```

Comment:
Can a common param be created and be referenced wherever needed?

Reply:
Creating a parameter instead of a string throws validation errors. I've reverted the changes 
![image](https://github.com/Azure/azure-rest-api-specs/assets/40211858/2271f77f-abb9-4140-aa5e-6afd9131dfd5)

Reply:
Please give this another try. You need to put this whole parameter object in the "parameters": { } section at the bottom and then reference it here using $ref: "parameters/<name>". It's just like any other $ref and it works. Look at the ApiVersionParameter reference in this same method for an example (ref is to an external file, but otherwise the same).

Reply:
I've given this another try. Unfortunately, after spending hours I am unable to find a good example or documentation on how this needs to be implemented to avoid validation errors. Unless this is a blocking change, or a working example is readily available, I request that we resolve this for the time being.